addons/mail/models/mail_activity_template.py

```diff
@@ -0,0 +1,101 @@
+                    raise ValidationError(
+                        _('The activity type "%(activity_type_name)s" is not compatible with the plan "%(plan_name)s"'
+                          ' because the plan is limited to some model not compatible with "%(activity_type_name)s"'
+                          ' (only applicable to "%(activity_type_model)s").',
+                          activity_type_name=record.activity_type_id.name, plan_name=record.plan_id.name,
+                          activity_type_model=record.activity_type_id.res_model))
+            else:
+                # Plan can be applied to any model -> check that the activity types is not restrained to a model
+                if record.activity_type_id.res_model:
+                    raise ValidationError(
+                        _('The activity type "%(activity_type_name)s" is not compatible with the plan "%(plan_name)s"'
+                          ' because "%(activity_type_name)s" can only be applied to the model "%(activity_type_model)s"'
+                          ' while the plan can be applied to any model. Please limit the plan to the model '
+                          '"%(activity_type_model)s".',
+                          activity_type_name=record.activity_type_id.name, plan_name=record.plan_id.name,
+                          activity_type_model=record.activity_type_id.res_model))
+
+    @api.constrains('responsible_id', 'responsible')
+    def _check_responsible(self):
+        """ Ensure that responsible_id is set when responsible is set to "other". """
+        for record in self:
+            if record.responsible == 'other' and not record.responsible_id:
+                raise ValidationError(_('When selecting responsible "other", you must specify a responsible.'))
+
+    @api.depends('activity_type_id')
+    def _compute_default_summary(self):
+        for plan_type in self:
+            if plan_type.activity_type_id and plan_type.activity_type_id.summary:
+                plan_type.summary = plan_type.activity_type_id.summary
+            else:
+                plan_type.summary = False
+
+    @api.depends('responsible')
+    def _compute_responsible_id(self):
+        for record in self:
+            if record.responsible != 'other' and record.responsible_id:
+                record.responsible_id = False
+
+    def _get_responsible_id(self, on_demand_responsible, record):
+        error = False
+        if self.responsible == 'other':
+            responsible = self.responsible_id
+        elif self.responsible == 'on_demand':
+            responsible = on_demand_responsible
+            if not responsible:
+                error = _('No responsible specified for %(activity_type_name)s: %(activity_summary)s.',
+                          activity_type_name=self.activity_type_id.name,
+                          activity_summary=self.summary or '-')
+        else:
+            raise ValueError(f'Invalid responsible value {self.responsible}.')
```

Comment:
If people are going to encounter this, should be an UserError (and tranlated). I suppose a misconfiguration could lead to this being raised ?